presqt/api_v1/views/resource.py

```diff
@@ -83,9 +85,109 @@ def get(self, request, target_name):
         # Fetch the target's resources
         try:
             resources = func(token)
-        except Exception as e:
+        except PresQTResponseException as e:
             # Catch any errors that happen within the target fetch
-            return Response(data={'error': e.__str__()}, status=status.HTTP_400_BAD_REQUEST)
+            return Response(data={'error': e.data}, status=e.status_code)
 
         serializer = ResourcesSerializer(instance=resources, many=True)
         return Response(serializer.data)
+
+
+class Resource(APIView):
+    """
+    **Supported HTTP Methods**
+
+    * GET: Retrieve a summary of the resource for the given Target that has been requested.
+    """
+    required_scopes = ['read']
+
+    def get(self, request, target_name, resource_id):
+        """
+        Retrieve details about a specific Resource.
+
+        Parameters
+        ----------
+        target_name : str
+            The name of the Target resource to retrieve.
+        resource_id : str
+            The id of the Resource to retrieve.
+
+        Returns
+        -------
+        200: OK
+        A dictionary like JSON representation of the requested Target resource.
+        {
+            "kind": "item",
+            "kind_name": "file",
+            "id": "5cd98518f244ec001ee8606b",
+            "title": "23296359282_934200ec59_o.jpg",
+            "date_created": "2019-05-13T14:54:17.129170Z",
+            "date_modified": "2019-05-13T14:54:17.129170Z",
+            "size": 1773294,
+            "hashes": {
+                "md5": "aaca7ef067dcab7cb8d79c36243823e4",
+                "sha256": "ea94ce54261720c16abb508c6dcd1fd481c30c09b7f2f5ab0b79e3199b7e2b55"
+            },
+            "extra": {
+                "last_touched": null,
+                "materialized_path": "/Images/23296359282_934200ec59_o.jpg",
+                "current_version": 1,
+                "provider": "osfstorage",
+                "path": "/5cd98518f244ec001ee8606b",
+                "current_user_can_comment": true,
+                "guid": null,
+                "checkout": null,
+                "tags": []
+            }
+        }
+
+        400: Bad Request
+        {
+            "error": "'new_target' does not support the action 'resource_collection'."
+        }
+
+        401: Unauthorized
+        {
+            "error": "'presqt-source-token' missing in the request headers."
+        }
+
+        403: Forbidden
+        {
+            "error": "User does not have access to this resource with the token provided."
+        }
+
+        404: Not Found
+        {
+            "error": "'bad_target' is not a valid Target name."
+        }
+        or
+        {
+            "error": "Resource with id 'bad_id' not found for this user."
+        }
+        """
+        action = 'resource_detail'
+
+        # Perform token validation
+        try:
+            token = token_validation(request)
```

Comment:
Curious... could token validation and target validation take place in the `func`?
Might that simplify this?